### ISSUE TITLE
Improve accessibility of form controls in HTML widgets

### ISSUE THREAD

**Issue by bmcfee** [matplotlib/matplotlib#17910]
### Bug report

MPL generates HTML with form controls in several places, eg, `animation.to_jshtml`, but the form controls are not generally accessible/screen-reader friendly. PR #15687 added "tooltip" to the js animation control widgets by using the `title` attribute, but this is not generally recommended by W3C, see: https://www.w3.org/WAI/tutorials/forms/labels/#using-the-title-attribute due to poor support from screen-readers.

The recommended approach is to use [ARIA attributes](https://www.w3.org/TR/WCAG20-TECHS/ARIA14.html).  There are several ways to do it: see [here](https://www.w3.org/WAI/tutorials/forms/labels/#hiding-label-text).  I'm not an expert on this stuff, but it seems like using the `aria-label=` attribute is appropriate here.  This [Mozilla developer page](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-label_attribute) suggests that `aria-label` should be used if there is no text in the control being annotated (which fits most of MPL's use cases, I think?), and otherwise to use `aria-labelledby`.

If people agree that this is a problem and with my suggested correction, I'm happy to open a PR.  However, I think it could probably use a round of discussion first.

**Code for reproduction**

n/a (not a bug)

**Actual outcome**

n/a

**Expected outcome**

n/a

**Matplotlib version**

  * Operating system: n/a
  * Matplotlib version: 3.2.2
  * Matplotlib backend (`print(matplotlib.get_backend())`): n/a (HTML)
  * Python version: 3.6.7
  * Jupyter version (if applicable): n/a
  * Other libraries: n/a

**Comment by dopplershift** [comment]
I'm not an expert on what the right attribute to use is, but I'm in favor generally of making the controls more accessible.

**Comment by timhoffm** [comment]
AFAICS the design pattern for AIRA-based tooltips is still work in progress: https://w3c.github.io/aria-practices/#tooltip

**Comment by bmcfee** [comment]
> AFAICS the design pattern for AIRA-based tooltips is still work in progress

Sure, but I think tooltips are a separate issue.  The `aria-label` attribute has no such disclaimer: https://w3c.github.io/aria-practices/#naming_with_aria-label

**Comment by timhoffm** [comment]
I'm confused. Does this mean we should put `aria-label` alongside with `title`?

Labeling as "good first issue":
- The main task is to dig out reference how the buttons should be attributed with `title` and/or `aria*` to
  - give a tooltip
  - make it accessible to screen readers
- Implementing is straight forward then

**Comment by bmcfee** [comment]
> Does this mean we should put `aria-label` alongside with `title`?

Yes, I think so.  Tooltips and screenreaders are related, but distinct functionality.



>     * The main task is to dig out reference how the buttons should be attributed

I think the links I provided in the issue do this, no?  Or do you mean something super-duper-official? :grin: 

Agreed that implementation will be easy here.  I wonder about other parts of the code that generate HTML though, eg `nbagg` backend or the `html5` animation output mode.  Are there other places where this pops up, or do these three cover everything?
